engineering/conventions-go.md

```diff
@@ -24,6 +24,10 @@ This guide documents development conventions for Go at source{d}. Check [general
 * If your project requires fetching or building dependencies, implement that in the `dependencies` rule.
 * If binaries can be built, do so in the `packages` rule. Note that [src-d/ci](https://github.com/src-d/ci) provides that rule by default.
 
+## Docker
+
+* In order to use alpine-based images, your binaries need to be built with `CGO_ENABLED=0`. [src-d/ci](https://github.com/src-d/ci/) does this by default. If your project uses `cgo`, you will have to use alpine for the build of binaries for Docker.
```

Comment:
I don't understand the last sentence.
Do you mean I can't use cross-compilation?

> If your project needs `cgo`, you will need to use alpine directly to build your binaries instead of cross-compilation.

Reply:
@campoy Right. This needs clarification.

You _can_ use cross-compilation, but if you build from Ubuntu, your cgo binaries will not work on an Alpine container (because they are compiled against glibc instead of musl). So your best options are either compiling from a `golang-alpine` container or using a `debian-slim` as base image for your Docker container.

Reply:
I suggest using always debian-slim, instead of alpine to avoid problems with musl.

Reply:
I.e https://github.com/src-d/gitbase-playground/pull/64 and same for bblfsh dashboard

Reply:
@mcuadros Maybe we should keep alpine where it is working properly already and not giving mainteinance problems?

Reply:
Any updates on this? I'll start changing driver build process to use `debian-slim` image. Alpine is adding more complexity because additional patches to Go are necessary to make it work.

Reply:
@dennwc  I think we still want to preserve Alpine of actual driver images, since it is much more convenient for users for these images to be small. `debian-slim` for build images should be ok if you are using `CGO_ENABLED=0`.

Reply:
We can keep alpine but for the next images, let's use debian-slim.

Reply:
I think the problem does not apply to bblfsh drivers anymore. The new build system based on multi-staged builds makes it easier to reuse multiple alpine images for the build and runtime.

cc @dennwc

Reply:
Correct, we don't have any problems with Alpine now. Although testing https://github.com/GoogleContainerTools/distroless as an alternative to Alpine will be interesting.